### PR TITLE
Test JuliaGPsDocs on repo

### PR DESCRIPTION
**Pull request by theogf**
Time for battle testing!

### CODE REVIEW

**Comment by theogf** [comment]
The errors seem unrelated to the PR...

**Comment by st--** [comment]
> The errors seem unrelated to the PR...

Yes, my sincere apologies! Turns out AbstractGPs 0.5.10 was breaking after all :disappointed:  @theogf once you've merged https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/pull/310 and https://github.com/JuliaGaussianProcesses/ApproximateGPs.jl/pull/125, the build issues should be solely due to your own mistakes again ;)

File: docs/make.jl [hunk 1]
```diff
@@ -58,11 +27,10 @@ makedocs(;
...
         "Home" => "index.md",
         "userguide.md",
-        "API" => ["api/index.md", "api/sparsevariational.md", "api/laplace.md"],
-        "Examples" =>
-            map(filter!(filename -> endswith(filename, ".md"), readdir(EXAMPLES_OUT))) do x
-                return joinpath("examples", x)
-            end,
+        "API" => joinpath.(Ref("api"), ["index.md", "sparsevariational.md", "laplace.md"]),
+        "Examples" => map(filter!(isdir, readdir("examples"))) do x
+            joinpath("examples", x, "example.md")
```
Comment (st--) [inline docs/make.jl:32]: why this change ? it seems a bit weird to have them all be called "example" ... :confused:

File: docs/make.jl [hunk 2]
```diff
@@ -58,11 +27,10 @@ makedocs(;
     pages=[
         "Home" => "index.md",
         "userguide.md",
-        "API" => ["api/index.md", "api/sparsevariational.md", "api/laplace.md"],
-        "Examples" =>
-            map(filter!(filename -> endswith(filename, ".md"), readdir(EXAMPLES_OUT))) do x
-                return joinpath("examples", x)
-            end,
+        "API" => joinpath.(Ref("api"), ["index.md", "sparsevariational.md", "laplace.md"]),
```
Comment (st--) [inline docs/make.jl:30]: why is the Ref() needed ?

**Review by st-- — APPROVED** [review]
Yay! looks great. The only thing I'd prefer changed is for it not to be called "example.md".... is it not possible to have it by name.md as before? if it's got to be hardcoded I'd prefer index.md ...

**Comment by theogf** [comment]
> Yay! looks great. The only thing I'd prefer changed is for it not to be called "example.md".... is it not possible to have it by name.md as before? if it's got to be hardcoded I'd prefer index.md ...

It is much easier to have a fixed name (this follow the same constraint as the "script.jl" thing. But sure we could go for another name

**Comment by theogf** [comment]
@devmotion was already using index.md, let's go with this then

**Comment by st--** [comment]
Seems to be working fine now. Great job @theogf  :tada:
